### ISSUE TITLE
How to generate multiple files based in a Json

### ISSUE THREAD

**Issue by fega** [shannonmoeller/gulp-hb#46]
Hello Shannon, I've been using gul-hb as an Static website generator.

Now, I want to generate multiple files based on a Json file array.

how can I achieve that?

thanks!

**Comment by shannonmoeller** [comment]
Glad this has been helpful to you. There are many ways you could generate multiple files, so I'll need a little more context. What does your array look like and what output are you trying to get from it?

**Comment by fega** [comment]
Ok!
my looks like:

```
{ "products" :[{
	"name" : "name of the product",
	"price" : "price of the product",
	"features" : ["feature of the product","feature of the product2"],
	"img" : "url"
}]
}
```

and I want to generate a html output like this
[http://ekus-group.github.io/productos/producto-test.html](http://ekus-group.github.io/productos/producto-test.html)

I want a file for every product in my array. 

thanks

**Comment by shannonmoeller** [comment]
@fega Sorry for the delay in replying. Generating files based on data objects gets a little more complicated as you have to create the virtual files manually. If you can, having the json as separate files is pretty slick:

https://github.com/shannonmoeller/tmp

Otherwise, something like this might work (untested; note the npm dependencies being imported):

```js
const File = require('vinyl');
const gulp = require('gulp');
const hb = require('gulp-hb');
const streamArray = require('stream-array');
const through = require('through2');

const products = [{
    "id": "foo",
    "name" : "name of the product",
    "price" : "price of the product",
    "features" : ["feature of the product","feature of the product2"],
    "img" : "url"
}];

function products() {
    return streamArray(products)
        .pipe(through(function (item, enc, cb) {
            const file = new File({
                path: `${item.id}.html`,
                data: item,
            });

            this.push(file);
            cb();
        }))
        .pipe(hb())
        .pipe(gulp.dest('dist'));
}

gulp.task('products', products);
```

**Comment by shannonmoeller** [comment]
Your milage may vary and this is just one of a few ways this could be tackled. Really depends on your specific needs.

**Comment by shannonmoeller** [comment]
Closing due to inactivity and that this isn't a bug-related thread. Feel free to reopen if needed.

**Comment by Poeticode** [comment]
Thought I might just post my problem here, since it's the same concept.

I've got a directory called 'dynamic' that has a bunch of json files. Each one is an array, containing info for a file. I'm wanting to iterate over the array in each JSON file, creating a file for each one using the template file they reference.

This is my current attempt:
http://pastebin.com/YKFANT0K

It seems to loop through each correctly. I tried using gutil.log() to check that it's reading everything correctly, and it seems to be all good in that regard. The problem is that it ends up only using the data from the last JSON file it reads. So if the directory has three JSON files, it loops through all three, but only creates the last one three times.
